static/src/javascripts/projects/common/modules/navigation/search.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+
+import bean from 'bean';
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+import config from 'lib/config';
+import throttle from 'lodash/functions/throttle';
+
+const focusSearchField = (): void => {
+    const $input = $('input.gsc-input');
+    if ($input.length > 0) {
+        $input.focus();
+    }
+};
+
+class Search {
+    gcsUrl: string;
+    resultSetSize: number;
+
+    constructor(): void {
+        const toggle = document.querySelector('.js-search-toggle');
+        let searchLoader;
+
+        if (
+            config.switches.googleSearch &&
+            config.page.googleSearchUrl &&
+            config.page.googleSearchId &&
+            toggle
+        ) {
+            this.gcsUrl = `${config.page.googleSearchUrl}?cx=${config.page
```

Comment:
We're querying the config for the same properties multiple times, in the conditional check that wraps this block and in this template string. Could we assign the values of these config properties to variables to avoid duplicating looking up the property.